google-cloud-dialogflow/MIGRATING.md

```diff
@@ -0,0 +1,274 @@
+regardless of whether they are required or optional. For example:
+
+Old:
+```
+client = Google::Cloud::Dialogflow::Sessions.new
+
+session = "projects/my-project/agent/sessions/my-session"
+query = { text: { text: "book a meeting room", language_code: "en-US" } }
+
+# Session and query are positional arguments
+response = client.detect_intent session, query
+```
+
+New:
+```
+client = Google::Cloud::Dialogflow.sessions
+
+session = "projects/my-project/agent/sessions/my-session"
+query = { text: { text: "book a meeting room", language_code: "en-US" } }
+
+# Session and query are keyword arguments
+response = client.detect_intent session: session, query_input: query
+```
+
+In the 1.0 release, it is also possible to pass a request object, either
+as a hash or as a protocol buffer.
+
+New:
+```
+client = Google::Cloud::Dialogflow.sessions
+
+request = Google::Cloud::Dialogflow::V2::DetectIntentRequest.new(
+  session: "projects/my-project/agent/sessions/my-session",
+  query_input: {
+    text: {
+      text: "book a meeting room",
+      language_code: "en-US"
+    }
+  }
+)
+
+# Pass a request object as a positional argument:
+response = client.detect_intent request
+```
+
+Finally, in older releases, to provide call options, you would pass a
+`Google::Gax::CallOptions` object with the `:options` keyword argument. In the
+1.0 release, pass call options using a _second set_ of keyword arguments.
+
+Old:
+```
+client = Google::Cloud::Dialogflow::Sessions.new
+
+session = "projects/my-project/agent/sessions/my-session"
+query = { text: { text: "book a meeting room", language_code: "en-US" } }
+
+options = Google::Gax::CallOptions.new timeout: 10_000
+
+response = client.detect_intent session, query, options: options
+```
+
+New:
+```
+client = Google::Cloud::Dialogflow.sessions
+
+session = "projects/my-project/agent/sessions/my-session"
+query = { text: { text: "book a meeting room", language_code: "en-US" } }
+
+# Use a hash to wrap the normal call arguments (or pass a request object), and
+# then add further keyword arguments for the call options.
+response = client.detect_intent(
+  { session: session, query_input: query },
+  timeout: 10_000
+)
+```
+
+### Resource Path Helpers
+
+The client library includes helper methods for the generating resource path
+strings passed to many calls. These helpers have changed in two ways:
+
+* In older releases, they are _class_ methods on the client class. In the 1.0
+  release, they are _instance_ methods on the client. They are also available
+  on a separate paths module that you can include elsewhere for convenience.
+* In older releases, arguments to a resource path helper are passed as
+  positional arguments. In the 1.0 release, they are passed as named keyword
+  arguments. Some helpers also cover several related paths, with different sets
+  of arguments. The class reference documentation provides details.
+
+Here is example usage under older releases:
```

Comment:
For consistency, I would use "Old:" and "New:" as in other example titles. And consider adding header styling: `##### Old:`

Reply:
Done.